crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1129,3 +1129,80 @@ fn is_entry_primitive_type<E: fmt::Debug, S: StorageView<E>>(
     }
     Ok(true)
 }
+
+pub fn special_argument_validate(
+    bytes: &[u8],
+    idx: u16,
+    layout: SpecialArgumentLayout,
+) -> Result<(), ExecutionError> {
+    bcs::from_bytes_seed(&layout, bytes).map_err(|_| {
+        ExecutionError::new_with_source(
+            ExecutionErrorKind::command_argument_error(CommandArgumentError::TypeMismatch, idx),
+            format!("Function expects {layout} but provided argument's value does not match",),
+        )
+    })
+}
+
+impl<'d> serde::de::DeserializeSeed<'d> for &SpecialArgumentLayout {
```

Comment:
@oxade if/when we want to add string limits, this would be the place. We could change the seed to have something that counts how many strings it has visited.

Reply:
Thanks for the callout.
While we're here, do you mind just implementing it? :) From our chats I thought the limits would come with this PR, which is why I punted my impl.

Reply:
I'm not longer sure we need it, but we can come back to it